interfaces/builtin/snap_interfaces_requests_control.go

```diff
@@ -35,13 +41,46 @@ const snapPromptingControlBaseDeclarationSlots = `
     deny-auto-connection: true
 `
 
+type requestsControlInterface struct {
+	commonInterface
+}
+
+func (iface *requestsControlInterface) BeforePreparePlug(plug *snap.PlugInfo) error {
+	// snaps can declare "handler-service" attribute on the plug, which
+	// indicates which user service will be handling prompt requests
+
+	handlerServiceAttr, isSet := plug.Attrs["handler-service"]
```

Comment:
Would it not be better to call this "user-handler-service" ?

Reply:
I thought of this but in principle is not impossible that we could have a system-wide service doing this in some scenarios, either way know which is which because the namespace is shared by all kind of services in the snap. So it's not clear there is a win with the longer name, especially as we expect for a while very few things consuming this interface